### PR TITLE
question 1.1

### PR DESCRIPTION
**Pull request by ashok-choudhary-0**

### CODE REVIEW

File: models/index.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const { Sequelize } = require('sequelize');
+
+const sequelize = new Sequelize('users', 'root', 'root123', {
```
Comment (amrit56789) [inline models/index.js:3]: instead of using sequelize details directly use env and move all the details in env file.

File: models/index.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  host: 'localhost',
+  dialect: 'mysql',
+  logging: false
+});
+
```
Comment (amrit56789) [inline models/index.js]: instead of using sequelize details directly use env and move all the details in env file.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+})
+
+app.use(bodyParser.json())  // middleware
+app.post("/role/add", userController.addRole);
+app.get("/role/list", userController.getAllRole)
+app.get("/role/list/:id", userController.getSingleRole)
+app.get("/role/list/:id", userController.getSingleRole)
+app.delete("/role/delete/:id", userController.deleteRole)
+app.put("/role/update/:id", userController.updateRole)
+
```
Comment (amrit56789) [inline index.js:8]: use seprate folder for apis.

File: index.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+require('dotenv').config();
+const port = process.env.PORT || 3000
+const User = require("./models/Roles")
+const bodyParser = require('body-parser')
+
+const userController = require("./controllers/userController")
+User.sync({ force: false });
+app.get("/", (req, res) => {
+  res.send("Hello world")
+})
```
Comment (amrit56789) [inline index.js]: is this api mentioned in the document?

File: controllers/userController.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
+const User = require("../models/Roles")
+
+
+const addRole = async (req, res) => {
+
+  try {
+    let data = req.body;
+    console.log(data)
```
Comment (amrit56789) [inline controllers/userController.js]: don't push console

File: controllers/userController.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  } catch (err) {
+    res.status(500).json(err);
+  }
+
+
+}
+
+const getAllRole = async (req, res) => {
+  try {
+    const data = await User.findAll({});
```
Comment (amrit56789) [inline controllers/userController.js]: take a meaningful variable name.

File: controllers/userController.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    res.status(200).json({ data: data })
+  } catch (err) {
+    res.status(500).json(err);
+  }
+}
+
+const getSingleRole = async (req, res) => {
+  try {
+    const data = await User.findOne({
```
Comment (amrit56789) [inline controllers/userController.js]: take a meaningful variable name.

File: controllers/userController.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    });
+    res.status(200).json({ data: data })
+  } catch (err) {
+    res.status(500).json(err);
+  }
+}
+
+const deleteRole = async (req, res) => {
+  try {
+    const data = await User.destroy({
```
Comment (amrit56789) [inline controllers/userController.js]: take a meaningful variable name.

File: controllers/userController.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    });
+    res.status(200).json({ data: data })
+  } catch (err) {
+    res.status(500).json(err);
+  }
+}
+const updateRole = async (req, res) => {
+  const updatedData = req.body
+  try {
+    const data = await User.update(updatedData, {
```
Comment (amrit56789) [inline controllers/userController.js]: take a meaningful variable name.

File: controllers/userController.js [hunk 10]
```diff
@@ -0,0 +1,68 @@
+const User = require("../models/Roles")
+
+
+const addRole = async (req, res) => {
+
+  try {
+    let data = req.body;
```
Comment (amrit56789) [inline controllers/userController.js]: also here choose meaningful variable name and try to use object destructuring.

File: controllers/userController.js [hunk 11]
```diff
@@ -0,0 +1,68 @@
+const User = require("../models/Roles")
+
```
Comment (amrit56789) [inline controllers/userController.js]: Remove white spaces.

File: controllers/userController.js (hunk 10 above)
Comment (amrit56789) [inline controllers/userController.js]: Choose meaningful variable name and also try to use destructuring.

File: controllers/userController.js (hunk 5 above)
Comment (amrit56789) [inline controllers/userController.js]: Don't push console

File: controllers/userController.js (hunk 6 above)
Comment (amrit56789) [inline controllers/userController.js]: Choose meaningful variable name.

File: controllers/userController.js (hunk 7 above)
Comment (amrit56789) [inline controllers/userController.js]: choose meaningful variable name.

File: controllers/userController.js (hunk 8 above)
Comment (amrit56789) [inline controllers/userController.js]: choose meaningful variable name.

File: index.js [hunk 12]
```diff
@@ -0,0 +1,26 @@
+const express = require("express");
+const app = express();
+require('dotenv').config();
+const port = process.env.PORT || 3000
+const User = require("./models/Roles")
+const bodyParser = require('body-parser')
+
+const userController = require("./controllers/userController")
+User.sync({ force: false });
+app.get("/", (req, res) => {
```
Comment (amrit56789) [inline index.js]: why you use this api? is this api mentioned in the document.

File: index.js [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+  res.send("Hello world")
+})
+
+app.use(bodyParser.json())  // middleware
+app.post("/role/add", userController.addRole);
+app.get("/role/list", userController.getAllRole)
+app.get("/role/list/:id", userController.getSingleRole)
+app.get("/role/list/:id", userController.getSingleRole)
+app.delete("/role/delete/:id", userController.deleteRole)
+app.put("/role/update/:id", userController.updateRole)
```
Comment (amrit56789) [inline index.js]: create a separate folder for all apis.

File: models/index.js (hunk 1 above)
Comment (amrit56789) [inline models/index.js]: use env file for sequelize details, don't push directly.

File: controllers/userController.js [hunk 14]
```diff
@@ -0,0 +1,56 @@
+const User = require("../models/Roles")
```
Comment (meanstack9716) [inline controllers/userController.js]: why User is coming from Roles model file? @ashok-choudhary-0

File: index.js [hunk 15]
```diff
@@ -0,0 +1,16 @@
+const express = require("express");
+const app = express();
+require('dotenv').config();
+const port = process.env.PORT || 3000
+const User = require("./models/Roles")
```
Comment (meanstack9716) [inline index.js]: same here @ashok-choudhary-0

File: index.js [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+const bodyParser = require('body-parser')
+const userRouter = require("./Router/userRouter")
```
Comment (meanstack9716) [inline index.js]: folder name should not be capital @ashok-choudhary-0

File: models/Roles.js [hunk 17]
```diff
@@ -0,0 +1,15 @@
+const { DataTypes } = require('sequelize');
+const sequelize = require('../dbConnection/connection')
+const User = sequelize.define('User', {
```
Comment (meanstack9716) [inline models/Roles.js]: The roles model table name is User? @ashok-choudhary-0

File: models/Roles.js [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+  name: {
+    type: DataTypes.STRING,
+    unique: true
```
Comment (meanstack9716) [inline models/Roles.js]: add not null as well as @ashok-choudhary-0

File: models/Roles.js [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+  },
+  description: {
+    type: DataTypes.STRING
+  }
+}, {
+  tableName: 'users',
```
Comment (meanstack9716) [inline models/Roles.js]: users???? it should be roles or role @ashok-choudhary-0

File: package.json [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+  "scripts": {
+    "test": "nodemon index.js"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "body-parser": "^1.20.2",
+    "dotenv": "^16.3.1",
+    "express": "^4.18.2",
+    "mongoose": "^7.5.1",
```
Comment (meanstack9716) [inline package.json]: why do you need mongooes in this? @ashok-choudhary-0

**Review by meanstack9716 — CHANGES_REQUESTED** [review]

**Pull request merged**